src/WebJobs.Script/Description/NodeFunctionInvoker.cs

```diff
@@ -116,24 +114,7 @@ internal NodeFunctionInvoker(ScriptHost host, BindingMetadata trigger, FunctionM
 
                 object functionResult = await ScriptFunc(scriptExecutionContext);
 
-                // normalize output binding results
-                IDictionary<string, object> functionOutputs = functionResult as IDictionary<string, object>;
-                if (functionResult != null && _outputBindings.Count == 1)
-                {
-                    // if there is only a single output binding allow that binding value
-                    // to be specified directly (i.e. normalize output format)
-                    var binding = _outputBindings.Single();
-                    functionOutputs = functionResult as IDictionary<string, object>;
-                    if (functionOutputs == null || !functionOutputs.ContainsKey(binding.Name))
-                    {
-                        functionOutputs = new Dictionary<string, object>()
-                        {
-                            { binding.Name, functionResult }
-                        };
-                    }
-                }
-
-                await ProcessOutputBindingsAsync(_outputBindings, input, binder, bindingData, functionOutputs);
+                await ProcessOutputBindingsAsync(_outputBindings, input, binder, bindingData, scriptExecutionContext, functionResult);
```

Comment:
So binding name is now required, right? Probably not a bad thing (outside of http case)